src/file-picker/src/FilePicker.js

```diff
@@ -184,7 +185,27 @@ FilePicker.propTypes = {
    * Class name passed to the FilePicker.
    * Only use this if you know what you are doing.
    */
-  className: PropTypes.string
+  className: PropTypes.string,
+
+  /**
+   * Text of the button when no file is selected
+   */
+  browseText: PropTypes.string,
```

Comment:
I think these props can probably be reduced based on the parts of the component we're rendering copy for - the button and the input value. 

How about an API that passes the current file count (for the button) and file list (for the input value)? This would require the consumer to set two props instead of four:

```tsx
interface FilePickerProps {
  // ...
  
  /**
   * Function that returns the call-to-action button text for selecting files
   */
  browseOrReplaceText?: (fileCount: number) => string;

  /**
   * Function that returns the text in the input field describing what file or files have been added
   */
  inputText?: (files: File[]) => string;

  // ...
}
```

The current logic in the component can be extracted into utility functions that are used when props aren't provided:

```tsx
const getBrowseOrReplaceText = (fileCount) => {
  const action = fileCount === 0 ? 'Select' : 'Replace'
  const fileLabel = fileCount > 1 ? 'files' : 'file'

  return `${action} ${fileLabel}`
}

const getInputValue = (files) => {
  if (files.length === 1) {
    return files[0].name
  }

  if (files.length > 1) {
    return `${files.length} files`
  }

  return ''
}
```

This also gives the consumer the flexibility to provide a custom input label based on the file name(s), if they want - which currently isn't possible.